agent/commands/run.go

```diff
@@ -57,27 +58,48 @@ func Run() {
 		cancel()
 	}()
 
-	var connectionUptimeService *connectionuptime.Service
+	var cfg config.Config
+	configFilepath, err := config.Get(&cfg, l)
+	if err != nil {
+		l.Fatalf("Failed to load configuration: %s.", err)
+	}
+
+	cleanupTmp(cfg.Paths.TempDir, l)
+	connectionUptimeService := connectionuptime.NewService(cfg.WindowConnectedTime)
+	connectionUptimeService.RunCleanupGoroutine(ctx)
+	supervisor := supervisor.NewSupervisor(ctx, &cfg.Paths, &cfg.Ports, &cfg.Server, &cfg.LogLinesCount)
+	supervisor.Run(ctx)
+	connectionChecker := connectionchecker.New(&cfg.Paths)
+	defaultsFileParser := defaultsfile.New()
+	v := versioner.New(&versioner.RealExecFunctions{})
+	r := runner.New(cfg.RunnerCapacity)
+	go r.Run(ctx)
+	client := client.New(&cfg, supervisor, r, connectionChecker, v, defaultsFileParser, connectionUptimeService, logStore)
+	localServer := agentlocal.NewServer(&cfg, supervisor, client, configFilepath, logStore)
+
+	go func() {
+		localServer.Run(ctx)
+		cancel()
+	}()
+
 	for {
-		cfg, configFilepath, err := config.Get(l)
+		_, err = config.Get(&cfg, l)
 		if err != nil {
 			l.Fatalf("Failed to load configuration: %s.", err)
 		}
-		config.ConfigureLogger(cfg)
+		config.ConfigureLogger(&cfg)
 		logStore.Resize(cfg.LogLinesCount)
 		l.Debugf("Loaded configuration: %+v", cfg)
 
-		cleanupTmp(cfg.Paths.TempDir, l)
-
 		logrus.Infof("Window check connection time is %.2f hour(s)", cfg.WindowConnectedTime.Hours())
-		if connectionUptimeService == nil {
-			connectionUptimeService = connectionuptime.NewService(cfg.WindowConnectedTime)
-			connectionUptimeService.RunCleanupGoroutine(ctx)
-		} else {
-			connectionUptimeService.SetWindowPeriod(cfg.WindowConnectedTime)
-		}
+		connectionUptimeService.SetWindowPeriod(cfg.WindowConnectedTime)
+
+		clientCtx, cancel := context.WithCancel(ctx)
+
+		_ = client.Run(clientCtx)
+		cancel()
```

Comment:
doesn't this call stop supervisor as well?

Reply:
No, it doesn't, the supervisor uses parental context and this call cancels only child context.